src/DataProtection/EntityFrameworkCore/src/EntityFrameworkCoreXmlRepository.cs

```diff
@@ -43,7 +43,10 @@ public virtual IReadOnlyCollection<XElement> GetAllElements()
             using (var scope = _services.CreateScope())
             {
                 var context = scope.ServiceProvider.GetRequiredService<TContext>();
-                return context.DataProtectionKeys.AsNoTracking().Select(key => TryParseKeyXml(key.Xml)).ToList().AsReadOnly();
+
+                // Put logger in a local such that `this` isn't captured.
+                var logger = _logger;
+                return context.DataProtectionKeys.AsNoTracking().Select(key => TryParseKeyXml(key.Xml, logger)).ToList().AsReadOnly();
```

Comment:
wouldn't it make more sense to put the select call **after** the "query"?
i.e. something like:

`context.DataProtectionKeys.AsNoTracking().AsEnumerable().Select(key => TryParseKeyXml(key.Xml, logger)).ToList().AsReadOnly()`
?

Reply:
@smitpatel @ajcvickers thoughts?

Reply:
In query with AsEnumerable, it would bring and materialize the DataProtectionKey object from server.
The version we have in place right now would only get value of Xml column and pass it to client function.
A better way to avoid client side evaluation in projection would be 
```C#
context.DataProtectionKeys.AsNoTracking().Select(key => key.Xml).AsEnumerable().Select(key => TryParseKeyXml(key, logger)).ToList().AsReadOnly()
```
Though in order to avoid that longer query client projection is the feature. I believe current query is the intended usage. No changes needed.
nit: Given that client side code is running `AsNoTracking` is redundant.